Source/ChilliSource/Rendering/Base/Renderer.cpp

```diff
@@ -43,55 +66,68 @@ namespace ChilliSource
     
     //------------------------------------------------------------------------------
     Renderer::Renderer() noexcept
-        : m_currentSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+        : m_currentMainSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+
     {
+
     }
     
     //------------------------------------------------------------------------------
     bool Renderer::IsA(InterfaceIDType interfaceId) const noexcept
     {
         return (Renderer::InterfaceID == interfaceId);
     }
+    
     //------------------------------------------------------------------------------
-    RenderSnapshot Renderer::CreateRenderSnapshot(const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
+    RenderSnapshot Renderer::CreateRenderSnapshot(const RenderTargetGroup* renderTarget, const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
     {
-        auto allocator = m_frameAllocatorQueue.Pop();
-        
-        return RenderSnapshot(allocator, resolution, clearColour, renderCamera);
+        return RenderSnapshot(renderTarget, resolution, clearColour, renderCamera);
     }
     
     //------------------------------------------------------------------------------
-    void Renderer::ProcessRenderSnapshot(RenderSnapshot renderSnapshot) noexcept
+    void Renderer::ProcessRenderSnapshots(IAllocator*&& frameAllocator, RenderSnapshot&& mainRenderSnapshot, std::vector<RenderSnapshot>&& offscreenRenderSnapshots) noexcept
```

Comment:
I would strongly recommend against this, it's a very unconventional pattern that is very likely to confuse people. It also doesn't stop then retaining a handle, only the handle they explicitly pass in. 

    auto obj = CreateNewObjectThatContainsAnAllocatorory(allocator);
    ProcessRenderSnapshots(allocator);
    obj.DoSomeAllocation(); // Will operate on a valid pointer.

It is a bit awkward that the allocator can be accessed from different threads, but it isn't user facing so not too much of a problem. In the future it's likely that the allocator passed down the pipeline will need to be thread-safe anyway in order to handle render command allocation. (i.e Command Lists allocate their own non-thread-safe fast linear allocators from the thread-safe linear allocator. When the list allocator is out of memory it requests more from the frame allocator. Should be super fast and thread safe)

If you do want to retain this, document it thoroughly.